apis/core/v1alpha1/helper/helpers.go

```diff
@@ -32,24 +39,29 @@ func SetOperation(obj *metav1.ObjectMeta, op v1alpha1.Operation) {
 	metav1.SetMetaDataAnnotation(obj, v1alpha1.OperationAnnotation, string(op))
 }
 
-// HasReconcileTimestampAnnotation checks if the obj has the given timeout annotation
-func HasReconcileTimestampAnnotation(obj metav1.ObjectMeta) bool {
-	_, ok := obj.Annotations[v1alpha1.ReconcileTimestampAnnotation]
+// HasTimestampAnnotation checks if the obj has the given timeout annotation
+func HasTimestampAnnotation(obj metav1.ObjectMeta, ta TimestampAnnotation) bool {
+	_, ok := obj.Annotations[string(ta)]
 	return ok
 }
```

Comment:
not sure if we need that function.
As you already pass a annotation there is no difference to the default `HasAnnotation` function?
(when calling the default function e.g. metav1.HasAnnotation(obj, v1alpha1.ReconcileTimestampAnnotation), you also directly see that you want the reconcile timestamp)